VitalHealthConnect/src/main/java/io/tryvital/vitalhealthconnect/records/RecordProcessor.kt

```diff
@@ -494,22 +437,60 @@ internal class HealthConnectRecordProcessor(
             ).toQuantitySample(currentDevice)
         }
 
-        val daySummariesByDate = awaitAll(*summaryAggregators).toMap()
-
-        val activities = (0 until numberOfDays).map { offset ->
-            val date = startDate.plusDays(offset.toLong())
-            Activity(
-                daySummary = daySummariesByDate[date],
-                activeEnergyBurned = activeEnergyBurnedByDate[date] ?: emptyList(),
-                basalEnergyBurned = basalMetabolicRateByDate[date] ?: emptyList(),
-                distanceWalkingRunning = distanceByDate[date] ?: emptyList(),
-                floorsClimbed = floorsClimbedByDate[date] ?: emptyList(),
-                steps = stepsByDate[date] ?: emptyList(),
-                vo2Max = vo2MaxByDate[date] ?: emptyList(),
-            )
-        }
+        val startTime = listOfNotNull(
```

Comment:
Day Summary has moved here, because now we need the timestamps from the samples to determine the range of Day Summaries to be recomputed.